cmd/src/code_intel_upload.go

```diff
@@ -132,12 +137,19 @@ func handleCodeIntelUpload(args []string) error {
 }
 
 // codeintelUploadOptions creates a set of upload options given the values in the flags.
-func codeintelUploadOptions(out *output.Output) upload.UploadOptions {
+func codeintelUploadOptions(out *output.Output, isSCIPAvailable bool) upload.UploadOptions {
 	var associatedIndexID *int
 	if codeintelUploadFlags.associatedIndexID != -1 {
 		associatedIndexID = &codeintelUploadFlags.associatedIndexID
 	}
 
+	cfg.AdditionalHeaders["Content-Type"] = "application/x-ndjson+lsif"
+	path := codeintelUploadFlags.uploadRoute
+	if isSCIPAvailable {
+		cfg.AdditionalHeaders["Content-Type"] = "application/x-protobuf+scip"
```

Comment:
Because https://github.com/sourcegraph/sourcegraph/pull/45269 has landed, you'll need to update your `sg/sg/lib` version in go.mod.